### PR TITLE
chore: add stylelint

### PR DESCRIPTION
**Pull request by ParsaArvanehPA**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [✔] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [] Tests for the changes have been added (for bug fixes / features)
- [] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[✔] Other... Please describe: Add stylelint
```

## What is the current behavior?
Issue Number: #8282 

## What is the new behavior?
•  Integrated stylelint with ant design v4 configurations.
•  Applied the same [plugins](https://github.com/ant-design/ant-design/blob/a52e205cef7d51e4b4b27a560569497ba5b6f899/package.json#L293-L298) and [rules](https://github.com/ant-design/ant-design/blob/4.x-stable/.stylelintrc.json) as ant v4 .
•  Modified the script section of package.json accordingly.
•  Resolved all stylelint linting errors.

## Does this PR introduce a breaking change?
```
[ ] Yes
[✔] No
```

## Other information
•  The dependencies are aligned with the same version as in ant v4, although newer versions are available.

•  The lint:style command in ant's package.json is stylelint '{site,components}/**/*.less', but the site/ directory is ignored by .gitignore, so only the components/ directory is linted by stylelint.

•  Appended npm run lint:style to the lint script, which enables stylelint to run in the lint pipeline.

### CODE REVIEW

**Comment by HyperLife1119** [comment]
So cool! I will check this out when I have time.

**Comment by Nicoss54** [comment]
LGTM

**Review by HyperLife1119 — APPROVED** [review]
LGTM

**Comment by ParsaArvanehPA** [comment]
Is that codefactor pipe failing ok? @HyperLife1119 @Nicoss54 

Oh and one more thing, remember you have to manually enable stylelint in your editors setting, in webstorm it's like:
image

**Comment by HyperLife1119** [comment]
No need to worry about codefactor.

**Comment by ParsaArvanehPA** [comment]
All the Npm jobs are giving this error:

image

Prior to this error, I was getting an error saying something about that I have to update my package.lock, so I deleted my package.lock, then ran `npm i` and committed the new package.lock; after that I'm getting this error.
Any ideas how to fix it?
@HyperLife1119 
@Nicoss54

**Comment by HyperLife1119** [comment]
It seems that due to re-running npm install, which caused us to upgrade to angular 17.2, the schematics test was wrong.  We need to upgrade to angular 17.2 in another PR and fix the schematics errors.

**Comment by HyperLife1119** [comment]
Ng-zorro has been updated to angular 17.2 via #8410 , you can merge the upstream latest commit and update package.lock again.

**Comment by ParsaArvanehPA** [comment]
> Ng-zorro has been updated to angular 17.2 via #8410 , you can merge the upstream latest commit and update package.lock again.

Yeah it worked. Thanks
@HyperLife1119 
@Nicoss54

**Comment by HyperLife1119** [comment]
Thank you for your amazing work, @ParsaArvanehPA :)

File: .stylelintrc.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    "custom-property-pattern": null,
+    "number-max-precision": 8,
+    "alpha-value-notation": "number",
+    "color-function-notation": "legacy",
+    "selector-class-pattern": null,
+    "selector-id-pattern": null,
+    "selector-not-notation": null,
+    "declaration-block-no-redundant-longhand-properties": null
+  },
+  "ignoreFiles": ["components/style/color/{bezierEasing,colorPalette,tinyColor}.less"]
```
Comment (HyperLife1119) [inline .stylelintrc.json:45]: We can also ignore `scripts/site/_site/**/*.less` to prevent them from affecting lint results.

Comment (ParsaArvanehPA) [reply .stylelintrc.json]: Absolutely right @HyperLife1119 
done.

Comment (HyperLife1119) [reply .stylelintrc.json]: Oh, and it looks like CodeFactor CI doesn't recognize stylelint's ignoreFiles configuration, so let's remove `scripts/site/_site/**/*.less` from ignoreFiles and format these .less files to fix CodeFactor CI's.

Comment (ParsaArvanehPA) [reply .stylelintrc.json:41]: @HyperLife1119 
All done; codefactor errors are gone. 
Thank you for putting so much time checking my PRs.

**Review by HyperLife1119 — APPROVED** [review]
LGTM

**Pull request merged**